### PR TITLE
Add support for core::f32/f64::min/max

### PR DESCRIPTION
**Pull request by lukfried**

### CODE REVIEW

File: prusti-viper/src/encoder/procedure_encoder.rs [hunk 1]
```diff
@@ -2174,6 +2178,78 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
...
+
+                            let lhs = vir::Expr::field(
+                                target_place.clone(),
+                                vir::Field::new("val_bool", vir::Type::Bool)
+                            );
+
+                            let operand = self.mir_encoder.encode_operand_expr(&args[0]).with_span(span)?;
+                            let expr = Expr::is_nan(operand);
+
+                            stmts.push(vir::Stmt::Assign(lhs, expr, vir::AssignKind::Ghost));
```
Comment (vakaras) [inline prusti-viper/src/encoder/procedure_encoder.rs]: The `vir::AssignKind::Ghost` looks wrong.

File: prusti-viper/src/encoder/procedure_encoder.rs [hunk 2]
```diff
@@ -2174,6 +2178,78 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
...
+                            let lhs = vir::Expr::field(
+                                target_place.clone(),
+                                vir::Field::new("val_floatf32", vir::Type::Float(FloatSize::F32))
+                            );
+                            let arg0 = self.mir_encoder.encode_operand_expr(&args[0]).with_span(span)?;
+                            let arg1 = self.mir_encoder.encode_operand_expr(&args[1]).with_span(span)?;
+                            
+                            let expr = Expr::min(arg0, arg1);
+
+                            stmts.push(vir::Stmt::Assign(lhs, expr, vir::AssignKind::Ghost));
```
Comment (vakaras) [inline prusti-viper/src/encoder/procedure_encoder.rs]: The `vir::AssignKind::Ghost` looks wrong.

File: prusti-viper/src/encoder/procedure_encoder.rs (hunk 1 above)
Comment (lukfried) [reply prusti-viper/src/encoder/procedure_encoder.rs]: I changed them to `vir::AssignKind::Copy`.

File: prusti-common/src/vir/optimizations/bitvector_analysis.rs [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+    continue_analysis: bool
+}
+
+
+
+impl BVAnalysis {
+
+    pub fn new(method: CfgMethod) -> Self {
+        BVAnalysis {
+            method: method.clone(),
```
Comment (vakaras) [inline prusti-common/src/vir/optimizations/bitvector_analysis.rs]: Why do you need to clone here?

File: prusti-common/src/vir/optimizations/bitvector_analysis.rs [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+            tainted_vars: HashSet::with_capacity(method.local_vars.len()),
+            continue_analysis: true,
+        }
+    }
+
+    pub fn get_new_method(&mut self) -> CfgMethod {
+        self.run();
+        return self.method.clone()
```
Comment (vakaras) [inline prusti-common/src/vir/optimizations/bitvector_analysis.rs]: Why do you need to clone here?

File: prusti-common/src/vir/optimizations/bitvector_analysis.rs [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+        Expr::Variant(e, _, _) |
+        Expr::Field(e, _, _) => {
+            vars.extend(vars_of_expr(e))
+        },
+        _ => ()
+    }
+    return vars
+}
+
+fn replace_tainted_vars_of_stmt(tainted_vars: &HashSet<LocalVar>, stmt: &Stmt) -> Stmt {
```
Comment (vakaras) [inline prusti-common/src/vir/optimizations/bitvector_analysis.rs]: Why are you not using a visitor here?

File: prusti-common/src/vir/ast/expr.rs [hunk 6]
```diff
@@ -292,7 +302,7 @@ impl fmt::Display for BinOpKind {
             BinOpKind::BitOr => write!(f, "|"),
             BinOpKind::BitXor => write!(f, "^"),
             BinOpKind::Shl => write!(f, "<<"),
-            BinOpKind::Shr => write!(f, ">>"),           
+            BinOpKind::LShr| BinOpKind::AShr => write!(f, ">>"),
```
Comment (vakaras) [inline prusti-common/src/vir/ast/expr.rs:305]: It is a bit confusing that two different operators are displayed in the same way. One of them could be `>>>` like in Java.

**Pull request closed without merging**